### ISSUE TITLE
Keydb memory issue

### ISSUE THREAD

**Issue by mykhailosmolynets** [JohnSully/KeyDB#188]
Keydb container uses almost 2x more memory from its **maxmemory** limit and it seems that doesn't evict keys at all. In some cases because of that it fails with OOM. There are a lot of keys to evict, but keydb skips it:
`db0:keys=343523,expires=343513,avg_ttl=153587`
As I can see in **INFO stats:**
```
expired_keys:31453912
expired_stale_perc:0.00
expired_time_cap_reached_count:0
evicted_keys:0
```
and according to documentation ( https://docs.keydb.dev/docs/commands/#notes-2 )in **evicted_keys** filed we should see at least some amount of evicted keys according to **maxmemory** policy. 

and **INFO memory**:
```
used_memory_human:3.50G
used_memory_rss_human:3.65G
used_memory_overhead:35864958
used_memory_peak_human:3.76G
maxmemory_human:2.00G
maxmemory_policy:volatile-lru
```

I have tried a couple of maxmemory policies, including **volatile-lru**, **volatile-ttl**, **volatile-lfu**, **allkeys-lfu**, but the result is the same. 

I'm using this chart for my infrastructure - https://github.com/Enapter/charts

Full **INFO memory** output
```
# Memory
used_memory:3909942320
used_memory_human:3.64G
used_memory_rss:4066144256
used_memory_rss_human:3.79G
used_memory_peak:4258918120
used_memory_peak_human:3.97G
used_memory_peak_perc:91.81%
used_memory_overhead:37605226
used_memory_startup:9107800
used_memory_dataset:3872337094
used_memory_dataset_perc:99.27%
allocator_allocated:3910962200
allocator_active:4032823296
allocator_resident:4147552256
total_system_memory:67557625856
total_system_memory_human:62.92G
used_memory_lua:40960
used_memory_lua_human:40.00K
used_memory_scripts:200
used_memory_scripts_human:200B
number_of_cached_scripts:1
maxmemory:2147483648
maxmemory_human:2.00G
maxmemory_policy:volatile-lru
allocator_frag_ratio:1.03
allocator_frag_bytes:121861096
allocator_rss_ratio:1.03
allocator_rss_bytes:114728960
rss_overhead_ratio:0.98
rss_overhead_bytes:-81408000
mem_fragmentation_ratio:1.04
mem_fragmentation_bytes:156221136
mem_not_counted_for_evict:0
mem_replication_backlog:1048576
mem_clients_slaves:17130
mem_clients_normal:980136
mem_aof_buffer:0
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0
```
Maybe there is some wrong configuration from my end, or understanding of some values, but I will really appreciate  if someone can help me to find the root cause of this issue.
**Image version** - eqalpha/keydb:x86_64_v5.3.2

**Comment by JohnSully** [comment]
Target: RC2

@mykhailosmolynets  I will try to replicate.  Based on the INFO MEMORY output this looks like a bug.

**Comment by mykhailosmolynets** [comment]
> Target: RC2
> 
> @mykhailosmolynets I will try to replicate. Based on the INFO MEMORY output this looks like a bug.

Thanks. Here is the list of non-default values that I unable in the chart, can be useful:
```
        --active-replica yes \
        --multi-master no \
        --maxmemory 2gb \
        --maxmemory-policy volatile-lru \
        --appendonly no \
        --proto-max-bulk-len 512mb \
        --client-query-buffer-limit 1gb \
        --bind 0.0.0.0 \
        --protected-mode no \
        --replica-serve-stale-data yes \
        --server-threads 2 
```

**Comment by JohnSully** [comment]
@mykhailosmolynets I've confirmed this repros on v5.3.2 and v5.3.3.  It does not occur with 6.0.8-RC which we just released yesterday.

**Comment by JohnSully** [comment]
Fixed in change: f112c77fcc3a60277ce344478bc37adb0fe4a99d

Closing as Fixed.  Do let me know if you have troubles on 6.0.8
